### PR TITLE
Throw error instead of assert in createContainer paths where empty summary tree is unsupported

### LINKED ISSUE #9039: blobs.spec.ts: Tests should not expect asserts. Likely the code path should throw a UsageError, rather than abuse assert
Tests should not expect asserts. Likely the code path should throw a UsageError, rather than abuse assert

_Originally posted by @anthony-murphy in https://github.com/microsoft/FluidFramework/pull/9025#r800979603_

### PR DESCRIPTION
**Pull request by justus-camp**
This throws an error instead of an assert in the case where an undefined summary tree is passed to drivers that don't support it. 

Fixes #9039

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/blobs.spec.ts [hunk 1]
```diff
@@ -266,7 +268,7 @@ describeNoCompat("blobs", (getTestObjectProvider) => {
         if (provider.driver.type !== "odsp") {
             // this flow is currently only supported on ODSP, the others should explicitly reject on attach
             return assert.rejects(attachP,
-                (err) => /(0x202)|(0x204)/.test(err.message) /* "create empty file not supported" */);
+                (err) => err.message === usageErrorMessage /* "create empty file not supported" */);
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/blobs.spec.ts:271]: nit (here and elsewhere):
```suggestion
                (err) => err.message === usageErrorMessage);
```

File: packages/drivers/local-driver/src/localDocumentServiceFactory.ts [hunk 2]
```diff
@@ -48,7 +47,9 @@ export class LocalDocumentServiceFactory implements IDocumentServiceFactory {
         clientIsSummarizer?: boolean,
     ): Promise<IDocumentService> {
         ensureFluidResolvedUrl(resolvedUrl);
-        assert(!!createNewSummary, 0x202 /* "create empty file not supported" */);
+        if (createNewSummary === undefined) {
+            throw new Error("Empty file summary creation isn't supported in this driver.");
```
Comment (markfields) [inline packages/drivers/local-driver/src/localDocumentServiceFactory.ts:51]: @ssimic2 - What was your follow-up about UsageError?  Whatever we decide from [here](https://github.com/microsoft/FluidFramework/pull/9650/files#r859003637) will also apply here I expect.

@justus-camp - This is good for now, but we might end up coming back later to change this to an error with a specific errorType (likely "usageError")

**Review by markfields — APPROVED** [review]
🎂 LGTM!

File: packages/drivers/local-driver/src/localDocumentServiceFactory.ts (hunk 2 above)
Comment (justus-camp) [reply packages/drivers/local-driver/src/localDocumentServiceFactory.ts:51]: Sounds good. I thought about implementing a new usageError type but used the generic for now. We can circle back around and add usageError if we find it necessary

**Pull request merged**